### PR TITLE
Add docs for backup of etcd in a cluster

### PR DESCRIPTION
**Pull request by foxylion**
ref #1506

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1511)

### CODE REVIEW

File: docs/etcd_backup.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+Kubernetes is relying on etcd for state storage. More details about the usage
+can be found [here](https://kubernetes.io/docs/admin/etcd/).
+
+## Backup requirement
+
+A Kubernetes cluster deployed with kops stores the etcd state in two different
+AWS EBS volumes for each master node. An EBS volume are designed to have a
+[failure rate](https://aws.amazon.com/ebs/details/#AvailabilityandDurability)
+of 0.1%-0.2% per year. This results in the requirement that these volumes should
+be backuped like any other persistent data.
```
Comment (justinsb) [inline docs/etcd_backup.md]: typo: "backed up"

File: docs/etcd_backup.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+## Restore volume backups
+
+In case of a lost EBS volume it is possible to restore the volume from a
+snapshot, more details [here](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-restoring-volume.html).
+
+Kubernetes uses protokube to identify the right volumes for etcd. Therefore it
+is important to tag the EBS volumes with the correct tags after restoring them
+from a EBS snapshot.
+
+protkube will look for the following tags:
```
Comment (justinsb) [inline docs/etcd_backup.md]: typo: "protokube"

File: docs/etcd_backup.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+* `KubernetesCluster` containing the cluster name (e.g. `k8s.mycompany.tld`)
+* `Name` containing the volume name (e.g. `eu-central-1a.etcd-main.k8s.mycompany.tld`)
+* `k8s.io/etcd/main` containg the availability zone of the volume (e.g. `
+eu-central-1a/eu-central-1a`)
```
Comment (justinsb) [inline docs/etcd_backup.md]: I think you need to not break the line here, it's confusing github anyway :-)

File: docs/etcd_backup.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# Backing up etcd
+
+Kubernetes is relying on etcd for state storage. More details about the usage
+can be found [here](https://kubernetes.io/docs/admin/etcd/).
```
Comment (chrislovecnm) [inline docs/etcd_backup.md]: Can we link etcd docs as well please

File: docs/etcd_backup.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+## Backup requirement
+
+A Kubernetes cluster deployed with kops stores the etcd state in two different
```
Comment (chrislovecnm) [inline docs/etcd_backup.md]: What are the to volumes used for. Please mention this is by master.

Comment (chrislovecnm) [inline docs/etcd_backup.md]: Scratch last sentence

File: docs/etcd_backup.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+AWS EBS volumes for each master node. An EBS volume are designed to have a
```
Comment (chrislovecnm) [inline docs/etcd_backup.md]: Two ebs volumes for each node.

File: docs/etcd_backup.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+Kubernetes does currently not provide any option to do regular backups of etcd
+out of the box.
+
+It is possible to setup a scheduled backup with EBS snapshots and AWS Lambda,
+more can be found [here](https://serverlesscode.com/post/lambda-schedule-ebs-snapshot-backups/).
+
+## Restore volume backups
+
+In case of a lost EBS volume it is possible to restore the volume from a
```
Comment (chrislovecnm) [inline docs/etcd_backup.md]: Make a note of the challenges of restoring from backup.

File: docs/etcd_backup.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+of 0.1%-0.2% per year. This results in the requirement that these volumes should
+be backuped like any other persistent data.
+
+## Create volume backups
+
+Kubernetes does currently not provide any option to do regular backups of etcd
+out of the box.
+
+It is possible to setup a scheduled backup with EBS snapshots and AWS Lambda,
+more can be found [here](https://serverlesscode.com/post/lambda-schedule-ebs-snapshot-backups/).
```
Comment (chrislovecnm) [inline docs/etcd_backup.md]: This is a second party site, can we get a tldr?

File: docs/etcd_backup.md (hunk 4 above)
Comment (foxylion) [inline docs/etcd_backup.md]: Do you have any suggestions which page would fit?

File: docs/etcd_backup.md (hunk 7 above)
Comment (foxylion) [inline docs/etcd_backup.md]: It tried to rewrite this section a bit. Do you like it now? :)

**Comment by foxylion** [comment]
@justinsb @chrislovecnm I updated the commit.

**Comment by chrislovecnm** [comment]
@k8s-bot ok to test

File: docs/etcd_backup.md (hunk 4 above)
Comment (chrislovecnm) [inline docs/etcd_backup.md]: Double check which version we are running ... https://coreos.com/etcd/docs/2.3.7/index.html <- 2.3.7 page

Comment (foxylion) [inline docs/etcd_backup.md]: I've added the link.

**Pull request merged**